charts/rollups-validator-node/templates/deployment.yaml

```diff
@@ -3,6 +3,93 @@
 SPDX-License-Identifier: Apache-2.0 (see LICENSE)
 */}}
 
+{{- if .Values.validator.localnode.enabled }}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "validator.fullname" . }}-localnode
+  labels: {{- include "validator.labels" . | nindent 4 }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      {{- include "validator.selectorLabels" . | nindent 6 }}
+      rollups.cartesi.io/component: localnode
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      {{- with .Values.validator.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "validator.selectorLabels" . | nindent 8 }}
+        rollups.cartesi.io/component: localnode
+    spec:
+      {{- with .Values.image.pullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "validator.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.validator.podSecurityContext | nindent 8 }}
+      initContainers:
+        - name: anvilstate
+          image: sunodo/rollups-node:0.2.0
+          command:
+            - cp
+            - /opt/cartesi/share/deployments/anvil_state.json
+            - /opt/cartesi/share/deployments/localhost.json
+            - /opt/anvil/
+          volumeMounts:
+            - name: blockchain-data
+              mountPath: /opt/anvil
+              readOnly: false
+      containers:
+        - name: anvil
+          image: {{ include "anvil.image" . }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          resources:
+            {{- toYaml .Values.validator.localnode.anvil.resources | nindent 12 }}
+          args:
+            - anvil
+            - --block-time
+            - "5"
+            - --load-state
+            - /opt/cartesi/share/deployments/anvil_state.json
```

Comment:
I think you already solved this on other resources, in case the user changes the `anvil.image` these args may fail on the custom image.

This args should be conditional.

Reply:
Fixed !